Data/Primitive/Array.hs

```diff
@@ -87,6 +89,18 @@ data Array a = Array
   { array# :: Array# a }
   deriving ( Typeable )
 
+#if MIN_VERSION_deepseq(1,4,3)
+instance NFData1 Array where
+  liftRnf r = Foldable.foldl' (\_ -> r) ()
+#endif
+
+instance NFData a => NFData (Array a) where
+#if MIN_VERSION_deepseq(1,4,3)
```

Comment:
There not a good reason to switch out the implementation of `rnf` based on the availability of `liftRnf`. Better would be to just unconditionally do `Foldable.foldl' (\_ -> rnf) ()`, or if the code duplication concerns you, factor out `Foldable.foldl' (\_ -> rnf) ()` into a top-level `liftRnfArray` function and just call that in both places.

Reply:
Sounds good to me.